src/main/java/appeng/thirdparty/codechicken/lib/model/Quad.java

```diff
@@ -0,0 +1,557 @@
+		 *
+		 * @param s The axis. side >> 1
+		 *
+		 * @return The y coord.
+		 */
+		public float dy( int s )
+		{
+			if( s > 0 )
+			{
+				return vec[1];
+			}
+			else
+			{
+				return vec[2];
+			}
+		}
+
+		/**
+		 * Interpolates the new color values for this Vertex using the others as a reference.
+		 *
+		 * @param interpHelper The InterpHelper to use.
+		 * @param others The other Vertices to use as a template.
+		 *
+		 * @return The same Vertex.
+		 */
+		public Vertex interpColorFrom( InterpHelper interpHelper, Vertex[] others )
+		{
+			for( int e = 0; e < 4; e++ )
+			{
+				float p1 = others[0].color[e];
+				float p2 = others[1].color[e];
+				float p3 = others[2].color[e];
+				float p4 = others[3].color[e];
+				//Only interpolate if colors are different.
+				if( p1 != p2 || p2 != p3 || p3 != p4 )
+				{
+					color[e] = interpHelper.interpolate( p1, p2, p3, p4 );
+				}
+			}
+			return this;
+		}
+
+		/**
+		 * Interpolates the new UV values for this Vertex using the others as a reference.
+		 *
+		 * @param interpHelper The InterpHelper to use.
+		 * @param others The other Vertices to use as a template.
+		 *
+		 * @return The same Vertex.
+		 */
+		public Vertex interpUVFrom( InterpHelper interpHelper, Vertex[] others )
+		{
+			for( int e = 0; e < 2; e++ )
+			{
+				float p1 = others[0].uv[e];
+				float p2 = others[1].uv[e];
+				float p3 = others[2].uv[e];
+				float p4 = others[3].uv[e];
+				if( p1 != p2 || p2 != p3 || p3 != p4 )
+				{
+					uv[e] = interpHelper.interpolate( p1, p2, p3, p4 );
+				}
+			}
+			return this;
+		}
+
+		/**
+		 * Interpolates the new LightMap values for this Vertex using the others as a reference.
+		 *
+		 * @param interpHelper The InterpHelper to use.
+		 * @param others The other Vertices to use as a template.
+		 *
+		 * @return The same Vertex.
+		 */
+		public Vertex interpLightMapFrom( InterpHelper interpHelper, Vertex[] others )
+		{
+			for( int e = 0; e < 2; e++ )
+			{
+				float p1 = others[0].lightmap[e];
+				float p2 = others[1].lightmap[e];
+				float p3 = others[2].lightmap[e];
+				float p4 = others[3].lightmap[e];
+				if( p1 != p2 || p2 != p3 || p3 != p4 )
+				{
+					lightmap[e] = interpHelper.interpolate( p1, p2, p3, p4 );
+				}
+			}
+			return this;
+		}
+
+		/**
+		 * Copies this Vertex to a new one.
+		 *
+		 * @return The new Vertex.
+		 */
+		public Vertex copy()
+		{
+			return new Vertex( this );
+		}
+
+		/**
+		 * Resets the Vertex to a new format. Expands the raw array if needed.
+		 *
+		 * @param format The format to reset to.
+		 */
+		public void reset( CachedFormat format )
+		{
+			//If the format is different and our raw array is smaller, then expand it.
+			if( !this.format.equals( format ) && format.elementCount > raw.length )
+			{
+				raw = new float[format.elementCount][4];
+			}
+			this.format = format;
+
+			vec = null;
+			normal = null;
+			color = null;
+			uv = null;
+			lightmap = null;
+
+			//            for (float[] f : raw) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)